pkg/metaserver/agent/metric/helper/system.go

```diff
@@ -23,78 +23,76 @@ import (
 		}
-		free = data.Value
-		_ = emitter.StoreFloat64(metricsNameNumaMetric, free, metrics.MetricTypeNameRaw,
-			metrics.ConvertMapToTags(map[string]string{
-				metricsTagKeyNumaID:     strconv.Itoa(numaID),
-				metricsTagKeyMetricName: consts.MetricMemFreeNuma,
-			})...)
-
-		data, err = metricsFetcher.GetNumaMetric(numaID, consts.MetricMemTotalNuma)
+		total, err = GetNumaMetric(metricsFetcher, emitter, consts.MetricMemTotalNuma, numaID)
 		if err != nil {
-			return 0, 0, 0, fmt.Errorf(errMsgGetNumaMetrics, consts.MetricMemTotalNuma, numaID, err)
+			return 0, 0, 0, fmt.Errorf(errMsgGetNumaMetrics, consts.MetricMemFreeNuma, numaID, err)
 		}
-		total = data.Value
-		_ = emitter.StoreFloat64(metricsNameNumaMetric, total, metrics.MetricTypeNameRaw,
-			metrics.ConvertMapToTags(map[string]string{
-				metricsTagKeyNumaID:     strconv.Itoa(numaID),
-				metricsTagKeyMetricName: consts.MetricMemTotalNuma,
-			})...)
 	} else {
-		data, err := metricsFetcher.GetNodeMetric(consts.MetricMemFreeSystem)
+		free, err = GetNodeMetric(metricsFetcher, emitter, consts.MetricMemFreeSystem)
 		if err != nil {
 			return 0, 0, 0, fmt.Errorf(errMsgGetSystemMetrics, consts.MetricMemFreeSystem, err)
 		}
-		free = data.Value
-		_ = emitter.StoreFloat64(metricsNameSystemMetric, free, metrics.MetricTypeNameRaw,
-			metrics.ConvertMapToTags(map[string]string{
-				metricsTagKeyMetricName: consts.MetricMemFreeSystem,
-			})...)
-
-		data, err = metricsFetcher.GetNodeMetric(consts.MetricMemTotalSystem)
+		total, err = GetNodeMetric(metricsFetcher, emitter, consts.MetricMemTotalSystem)
 		if err != nil {
 			return 0, 0, 0, fmt.Errorf(errMsgGetSystemMetrics, consts.MetricMemTotalSystem, err)
 		}
-		total = data.Value
-		_ = emitter.StoreFloat64(metricsNameSystemMetric, total, metrics.MetricTypeNameRaw,
-			metrics.ConvertMapToTags(map[string]string{
-				metricsTagKeyMetricName: consts.MetricMemTotalSystem,
-			})...)
 	}
 
-	data, err := metricsFetcher.GetNodeMetric(consts.MetricMemScaleFactorSystem)
+	scaleFactor, err = GetNodeMetric(metricsFetcher, emitter, consts.MetricMemScaleFactorSystem)
 	if err != nil {
 		return 0, 0, 0, fmt.Errorf(errMsgGetSystemMetrics, consts.MetricMemScaleFactorSystem, err)
 	}
-	scaleFactor = data.Value
-	_ = emitter.StoreFloat64(metricsNameSystemMetric, scaleFactor, metrics.MetricTypeNameRaw,
-		metrics.ConvertMapToTags(map[string]string{
-			metricsTagKeyMetricName: consts.MetricMemScaleFactorSystem,
-		})...)
 
 	return free, total, scaleFactor, nil
 }
 
 func GetSystemKswapdStealMetrics(metricsFetcher metric.MetricsFetcher, emitter metrics.MetricEmitter) (metricutil.MetricData, error) {
-	kswapdSteal, err := metricsFetcher.GetNodeMetric(consts.MetricMemKswapdstealSystem)
+	return GetNodeMetricWithTime(metricsFetcher, emitter, consts.MetricMemKswapdstealSystem)
+}
+
+func GetNodeMetricWithTime(metricsFetcher metric.MetricsFetcher, emitter metrics.MetricEmitter, metricName string) (metricutil.MetricData, error) {
```

Comment:
`GetNodeMetricWithTime` is kind of weird, why node needs a function like this while numa not

Reply:
It depends if there is a caller who needs timestamp of metric. For now, only node metrics faces this kind of need, so I just provide this kind of function for node metrics. I'll provide one for numa metrics to make them looks more consistent.